File: src/server/gkill/api/req_res/submit_kftl_text_request.go
package req_res

type SubmitKFTLTextRequest struct {
	SessionID  string `json:"session_id"`
	KFTLText   string `json:"kftl_text"`
	LocaleName string `json:"locale_name"`
	// IdempotencyKey は同じ送信の再配送を1回の登録に畳むための任意キー。
	// 空なら冪等判定をしない（従来どおり毎回登録）。Wear のワーカー再送で
	// 結果だけ届かなかったとき、同じキーで再送しても二重登録にならない（監査 S3-wear）。
	IdempotencyKey string `json:"idempotency_key,omitempty"`
}

File: src/client/classes/api/req_res/submit-kftl-text-request.ts
'use strict'

import { GkillAPIRequest } from '../gkill-api-request'

// メモ帳（KFTL）のテキストをサーバに解釈・記録させる要求（Go 側 req_res.SubmitKFTLTextRequest）。
// 解釈と書き込みはサーバの1実装だけで行う（ADR-0507）。Wear OS / MCP と同じ入口。
export class SubmitKFTLTextRequest extends GkillAPIRequest {

    kftl_text: string = ""

    // 同じ送信の再配送を1回の登録に畳むためのキー。送信のたびに新しい UUID を採る
    // （内容ハッシュにしない ―― 意図した同一内容の再送が畳まれて記録できなくなる）
    idempotency_key: string = ""

    constructor() {
        super()
    }

}
